### PR TITLE
Add support for passing configuration to custom AccessTokenProvider

### PR DESCRIPTION
**Pull request by tom-s-powell**
A replacement for https://github.com/GoogleCloudDataproc/spark-bigquery-connector/pull/811. The goal here is to let a custom `AccessTokenProvider` be passed optional configuration via read options.

### CODE REVIEW

File: bigquery-connector-common/src/test/java/com/google/cloud/bigquery/connector/common/integration/CustomCredentialsIntegrationTest.java
Comment (davidrabinowitz) [inline bigquery-connector-common/src/test/java/com/google/cloud/bigquery/connector/common/integration/CustomCredentialsIntegrationTest.java:84]: The test is not giving a valid access token, but a dummy string. This will cause the call to the BigQuery API to fail. I suggest that the field name to be "config" and to check that its value was properly set. The `getAccessToken()` implementation can remain the same.

**Review by davidrabinowitz — CHANGES_REQUESTED** [review]

**Comment by davidrabinowitz** [comment]
Please run `./mvnw spotless:apply -Pall` before pushing the next commit.

File: README.md
Comment (tom-s-powell) [reply README.md]: Ah hadn't realised this. Fixed that up.

**Comment by tom-s-powell** [comment]
Appreciate the review @davidrabinowitz. I've addressed those comments, hope the test now looks as you were expecting? I've run `./mvnw spotless:apply -Pall` so hopefully looks okay.

File: bigquery-connector-common/src/test/java/com/google/cloud/bigquery/connector/common/integration/CustomCredentialsIntegrationTest.java
Comment (davidrabinowitz) [inline bigquery-connector-common/src/test/java/com/google/cloud/bigquery/connector/common/integration/CustomCredentialsIntegrationTest.java:90]: Expected value is 1, as `getAccessToken()` has been called once in order to make the call the BigQuery API

**Review by davidrabinowitz — CHANGES_REQUESTED** [review]

File: bigquery-connector-common/src/test/java/com/google/cloud/bigquery/connector/common/integration/CustomCredentialsIntegrationTest.java
Comment (tom-s-powell) [reply bigquery-connector-common/src/test/java/com/google/cloud/bigquery/connector/common/integration/CustomCredentialsIntegrationTest.java]: Fixed that

**Comment by tom-s-powell** [comment]
@davidrabinowitz couldn't quite tell, but the failing tests don't seem to be related to this change?

**Comment by davidrabinowitz** [comment]
> @davidrabinowitz couldn't quite tell, but the failing tests don't seem to be related to this change?

Yes, we have a flaky test in the past few days. I've `@Ignore`d it in PR #917, but that's waiting for review. I'll keep an eye on this and will merge once the test passes

**Comment by tom-s-powell** [comment]
Great thanks very much.

**Pull request merged**
